planning/route_handler/src/route_handler.cpp

```diff
@@ -715,12 +715,28 @@ bool RouteHandler::getClosestLaneletWithinRoute(
 bool RouteHandler::getNextLaneletWithinRoute(
   const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * next_lanelet) const
 {
+  lanelet::ConstLanelet start_lanelet;
+  lanelet::ConstLanelet goal_lanelet;
+  bool flag_check = true;
+  if (
+    !getGoalLanelet(&goal_lanelet) ||
+    !getClosestLaneletWithinRoute(route_msg_.start_pose, &start_lanelet)) {
+    flag_check = false;
+  }
+  if (goal_lanelet.id() == start_lanelet.id()) {
+    flag_check = false;
+  }
   if (exists(goal_lanelets_, lanelet)) {
     return false;
   }
   const auto following_lanelets = routing_graph_ptr_->following(lanelet);
   for (const auto & llt : following_lanelets) {
     if (exists(route_lanelets_, llt)) {
+      if (flag_check) {
```

Comment:
if possible, it is nicer to reduce the number of nested if, so i would suggest the following..
```
if(!(flag_check && (start_lanelet.id() == llt.id())){
*next_lanelet = llt;
return true;
}
```
might be more simpler?
this reduces nested

Reply:
Yes, I will fix it